py/src/ai/h2o/sparkling/Initializer.py

```diff
@@ -79,11 +84,11 @@ def __add_sparkling_jar_to_spark(sc):
         sc._jsc.addJar(sw_jar_file)
 
     @staticmethod
-    def __extracted_jar_path(sc):
+    def __extracted_jar_path():
 
         if Initializer.__extracted_jar_dir is None:
             zip_file = Initializer.__get_pysparkling_zip_path()
-            Initializer.__extracted_jar_dir = sc._temp_dir
+            Initializer.__extracted_jar_dir = tempfile.mkdtemp()
```

Comment:
Is this temp dir deleted somewhere?

Reply:
Good catch, nope. But I can actually revert back and use the spark temp dir

Reply:
I will use the spark temp dir in the original use-case but will create new temp dir only in this specific use-case and mark it for deletion when the app ends. Just to make sure we do not affect existing users